### PR TITLE
Remove unneeded checks before `delete`

### PR DESCRIPTION
**Pull request by kunaltyagi**

### CODE REVIEW

**Comment by kunaltyagi** [comment]
> Your first commit includes some unrelated index_t changes

Oops, I based this off a wrong branch, not master. Will correct that.

> please don't commit trailing space stripping changes that bloat the diff.

Editors love stripping whitespace. I'll update the patch.

**Comment by taketwo** [comment]
Oh, forgot to write, I prefer the second method.

File: common/include/pcl/common/impl/bivariate_polynomial.hpp [hunk 1]
```diff
@@ -109,12 +109,12 @@ pcl::BivariatePolynomialT<real>::deepCopy (const pcl::BivariatePolynomialT<real>
     degree = other.degree;
     parameters = new real[getNoOfParameters ()];
   }
-  if (other.gradient_x == NULL)
+  if (!other.gradient_x)
   {
-    delete gradient_x; gradient_x=NULL;
-    delete gradient_y; gradient_y=NULL;
+    delete gradient_x;
+    delete gradient_y;
```
Comment (SunBlack) [inline common/include/pcl/common/impl/bivariate_polynomial.hpp:115]: I believe yo should still set the variables to `nullptr` after this, as the object still exists.

File: common/src/gaussian.cpp [hunk 2]
```diff
@@ -183,37 +182,39 @@ pcl::GaussianKernel::convolveCols (const pcl::PointCloud<float>& input,
...
       output (i,j) = 0;
       for (int k = static_cast<int>(kernel_width), l = static_cast<int>(j - radius) ; k >= 0 ; k--, l++)
       {
-        output (i,j) += (*input_) (i,l) * kernel[k];
+        output (i,j) += (*unaliased_input) (i,l) * kernel[k];
       }
     }
 
-    for ( ; j < input_->height ; j++)
+    for ( ; j < unaliased_input->height ; j++)
```
Comment (SunBlack) [inline common/src/gaussian.cpp:217]: 1) What is the adjusted code doing?
2) Isn't it the same code as 50 lines above? Could you do code deduplication?

File: filters/include/pcl/filters/impl/conditional_removal.hpp [hunk 3]
```diff
@@ -90,11 +90,8 @@ pcl::FieldComparison<PointT>::FieldComparison (
 template <typename PointT>
 pcl::FieldComparison<PointT>::~FieldComparison () 
 {
-  if (point_data_ != nullptr)
-  {
-    delete point_data_;
-    point_data_ = nullptr;
-  }
+  delete point_data_;
+  point_data_ = nullptr;
```
Comment (SunBlack) [inline filters/include/pcl/filters/impl/conditional_removal.hpp:94]: As you deleted `point_data_` within the destructor there is no need to set it to `nullptr` after this.

Comment (kunaltyagi) [reply filters/include/pcl/filters/impl/conditional_removal.hpp:94]: Yeah, I didn't check where I'm making the edits 😄  Will remove it soon

File: common/src/gaussian.cpp (hunk 2 above)
Comment (kunaltyagi) [reply common/src/gaussian.cpp:217]: 1. no need of manually managing the local memory by conditionally creating a new memory block to unalias input and output pointers
2. Almost. And yes. But maybe not in this PR

**Comment by kunaltyagi** [comment]
It's a bit harder to see what's wrong in the checks sidebar with the new CI script

File: doc/tutorials/content/sources/iros2011/src/openni_capture.cpp [hunk 4]
```diff
@@ -24,7 +24,6 @@ OpenNICapture::~OpenNICapture ()
 {
   // Stop the grabber when shutting down
   grabber_.stop ();
-  if (preview_)
     delete preview_;
```
Comment (SergioRAgostinho) [inline doc/tutorials/content/sources/iros2011/src/openni_capture.cpp:27]: Ind.

File: gpu/octree/src/octree.cpp [hunk 5]
```diff
@@ -78,7 +78,6 @@ pcl::gpu::Octree::~Octree() { clear(); }
 
 void pcl::gpu::Octree::clear()
 {
-    if (impl)
         delete static_cast<OctreeImpl*>(impl);
```
Comment (SergioRAgostinho) [inline gpu/octree/src/octree.cpp:81]: Ind.

File: keypoints/include/pcl/keypoints/impl/iss_3d.hpp [hunk 6]
```diff
@@ -235,13 +235,11 @@ pcl::ISSKeypoint3D<PointInT, PointOutT, NormalT>::initCompute ()
     return (false);
   }
 
-  if (third_eigen_value_)
     delete[] third_eigen_value_;
```
Comment (SergioRAgostinho) [inline keypoints/include/pcl/keypoints/impl/iss_3d.hpp:238]: Ind.

File: keypoints/include/pcl/keypoints/impl/iss_3d.hpp [hunk 7]
```diff
@@ -235,13 +235,11 @@ pcl::ISSKeypoint3D<PointInT, PointOutT, NormalT>::initCompute ()
...
 
   third_eigen_value_ = new double[input_->size ()];
   memset(third_eigen_value_, 0, sizeof(double) * input_->size ());
 
-  if (edge_points_)
     delete[] edge_points_;
```
Comment (SergioRAgostinho) [inline keypoints/include/pcl/keypoints/impl/iss_3d.hpp:243]: Ind.

File: recognition/include/pcl/recognition/ransac_based/bvh.h [hunk 8]
```diff
@@ -150,13 +150,10 @@ namespace pcl
             }
 
             virtual ~Node ()
-            {
-              if ( children_[0] )
               {
                 delete children_[0];
                 delete children_[1];
               }
```
Comment (SergioRAgostinho) [inline recognition/include/pcl/recognition/ransac_based/bvh.h:156]: Ind.

File: recognition/include/pcl/recognition/ransac_based/bvh.h [hunk 9]
```diff
@@ -249,13 +246,10 @@ namespace pcl
         /** \brief Frees the memory allocated by this object. After that, you have to call build to use the tree again. */
         void
         clear()
-        {
-          if ( root_ )
           {
             delete root_;
             root_ = nullptr;
           }
```
Comment (SergioRAgostinho) [inline recognition/include/pcl/recognition/ransac_based/bvh.h:252]: Ind.

File: recognition/include/pcl/recognition/ransac_based/orr_octree.h [hunk 10]
```diff
@@ -241,23 +241,17 @@ namespace pcl
 
             inline void
             deleteChildren ()
-            {
-              if ( children_ )
               {
                 delete[] children_;
                 children_ = nullptr;
               }
```
Comment (SergioRAgostinho) [inline recognition/include/pcl/recognition/ransac_based/orr_octree.h:247]: Ind.

File: recognition/include/pcl/recognition/ransac_based/orr_octree.h [hunk 11]
```diff
@@ -241,23 +241,17 @@ namespace pcl
...
-            }
 
             inline void
             deleteData ()
-            {
-              if ( data_ )
               {
                 delete data_;
                 data_ = nullptr;
               }
```
Comment (SergioRAgostinho) [inline recognition/include/pcl/recognition/ransac_based/orr_octree.h:254]: Ind.

File: recognition/src/ransac_based/orr_octree_zprojection.cpp [hunk 12]
```diff
@@ -60,7 +60,6 @@ pcl::recognition::ORROctreeZProjection::clear ()
     {
       // Delete pixel by pixel
       for ( int j = 0 ; j < num_pixels_y_ ; ++j )
-        if ( pixels_[i][j] )
           delete pixels_[i][j];
```
Comment (SergioRAgostinho) [inline recognition/src/ransac_based/orr_octree_zprojection.cpp:63]: Ind.

File: recognition/src/ransac_based/orr_octree_zprojection.cpp [hunk 13]
```diff
@@ -76,7 +75,6 @@ pcl::recognition::ORROctreeZProjection::clear ()
     for ( int i = 0 ; i < num_pixels_x_ ; ++i )
     {
       for ( int j = 0 ; j < num_pixels_y_ ; ++j )
-        if ( sets_[i][j] )
           delete sets_[i][j];
```
Comment (SergioRAgostinho) [inline recognition/src/ransac_based/orr_octree_zprojection.cpp:78]: Ind.

File: surface/include/pcl/surface/3rdparty/poisson4/mat.hpp [hunk 14]
```diff
@@ -43,10 +43,8 @@ namespace pcl
     template <class Real>
     MinimalAreaTriangulation<Real>::~MinimalAreaTriangulation(void)
     {
-      if(bestTriangulation)
         delete[] bestTriangulation;
```
Comment (SergioRAgostinho) [inline surface/include/pcl/surface/3rdparty/poisson4/mat.hpp:46]: Ind.

File: surface/include/pcl/surface/3rdparty/poisson4/mat.hpp [hunk 15]
```diff
@@ -43,10 +43,8 @@ namespace pcl
...
       bestTriangulation=NULL;
-      if(midPoint)
         delete[] midPoint;
```
Comment (SergioRAgostinho) [inline surface/include/pcl/surface/3rdparty/poisson4/mat.hpp:48]: Ind.

File: surface/include/pcl/surface/3rdparty/poisson4/mat.hpp [hunk 16]
```diff
@@ -104,9 +102,7 @@ namespace pcl
         }
         return;
       }
-      if(bestTriangulation)
         delete[] bestTriangulation;
```
Comment (SergioRAgostinho) [inline surface/include/pcl/surface/3rdparty/poisson4/mat.hpp:105]: Ind.

File: surface/include/pcl/surface/3rdparty/poisson4/mat.hpp [hunk 17]
```diff
@@ -104,9 +102,7 @@ namespace pcl
...
-      if(midPoint)
         delete[] midPoint;
```
Comment (SergioRAgostinho) [inline surface/include/pcl/surface/3rdparty/poisson4/mat.hpp:106]: Ind.

File: surface/include/pcl/surface/3rdparty/poisson4/mat.hpp [hunk 18]
```diff
@@ -123,9 +119,7 @@ namespace pcl
     template <class Real>
     Real MinimalAreaTriangulation<Real>::GetArea(const std::vector<Point3D<Real> >& vertices)
     {
-      if(bestTriangulation)
         delete[] bestTriangulation;
```
Comment (SergioRAgostinho) [inline surface/include/pcl/surface/3rdparty/poisson4/mat.hpp:122]: Ind.

File: surface/include/pcl/surface/3rdparty/poisson4/mat.hpp [hunk 19]
```diff
@@ -123,9 +119,7 @@ namespace pcl
...
-      if(midPoint)
         delete[] midPoint;
```
Comment (SergioRAgostinho) [inline surface/include/pcl/surface/3rdparty/poisson4/mat.hpp:123]: Ind.

**Comment by kunaltyagi** [comment]
Apps has 3 PR for formatting, I'll let them touch it.

**Pull request merged**